starlette/staticfiles.py

```diff
@@ -73,19 +76,19 @@ def __init__(self, scope, path, check_directory=None):
         if not (stat.S_ISDIR(stat_result.st_mode) or stat.S_ISLNK(stat_result.st_mode)):
             raise RuntimeError("StaticFiles path '%s' is not a directory." % directory)
 
-    async def __call__(self, receive, send):
+    async def __call__(self, receive: Receive, send: Send) -> None:
         if self.check_directory is not None:
             await self.check_directory_configured_correctly()
 
         try:
             stat_result = await aio_stat(self.path)
         except FileNotFoundError:
-            response = PlainTextResponse("Not Found", status_code=404)
+            response = PlainTextResponse("Not Found", status_code=404)  # type: Response
```

Comment:
I think this is the only thing I’d like to change. Ideally I think we should ensure the our function definitions are entirely typed, but I’m not convinced I like the variable typing, as it’s quite noisy.

Either we could just remove the variable typing, or else we could add a `response = None  # Type: Response` to the top of the method in cases like this?

Reply:
I agree. I am not a big fan of it as well. We can not do `response = None # type: Response` because we will have an error, but we can do `response = None # type:  typing.Any`

Another solution could be rethinking a little bit this block of code and declare `response = PlainTextResponse("Not Found", status_code=404)  # type: Response` since this is should be de default value.